ixmp/core/timeseries.py

```diff
@@ -221,10 +235,13 @@ def transact(self, message: str = "", condition: bool = True):
         >>> # Changes to `ts` have been committed
         """
         # TODO implement __enter__ and __exit__ to allow simpler "with ts: …"
+        from ixmp.utils import discard_on_error as doe
+
         if condition:
             maybe_check_out(self)
         try:
-            yield
+            with doe(self) if discard_on_error else nullcontext():
```

Comment:
suggest to simply spell out what `doe` means for future developers (ie not import as), recognizing that the argument has the same name

Reply:
Used a longer local name and added an explanatory comment.